imap_processing/hit/l1a/hit_l1a.py

```diff
@@ -0,0 +1,208 @@
+        Memory dump
+    """
+
+    HIT_AUT = 1250  # Autonomy
+    HIT_HSKP = 1251  # Housekeeping
+    HIT_SCIENCE = 1252  # Science
+    HIT_IALRT = 1253  # I-ALiRT
+    HIT_MEMDUMP = 1255  # Memory dump
+
+
+def hit_l1a_data(packet_file: str):
+    """
+    Process HIT L0 data into L1A data products.
+
+    Parameters
+    ----------
+    packet_file : str
+        Path to the CCSDS data packet file.
+
+    Returns
+    -------
+    dict
+        List of file paths to CDF data product files
+    """
+    # Decom, sort, and group packets by apid
+    packets = decom_packets(packet_file)
+    sorted_packets = sorted(packets, key=lambda x: x.data["SHCOARSE"].derived_value)
+    grouped_data = group_data(sorted_packets)
+
+    # Create datasets
+    # TODO define keys to skip for each apid. Currently just have
+    #  a list for housekeeping.
+    skip_keys = [
+        "shcoarse",
+        "ground_sw_version",
+        "packet_file_name",
+        "ccsds_header",
+        "leak_i_raw",
+    ]
+    datasets = create_datasets(grouped_data, skip_keys)
+
+    # Create CDF files
+    logger.info("Creating CDF files for HIT L1A data")
+    cdf_filepaths = []
+    for dataset in datasets.values():
+        cdf_file = write_cdf(dataset)
+        cdf_filepaths.append(cdf_file)
+    logger.info("L1A CDF files created")
+    return cdf_filepaths
+
+
+def decom_packets(packet_file: str):
+    """
+    Unpack and decode packets using CCSDS file and XTCE packet definitions.
+
+    Parameters
+    ----------
+    packet_file : str
+        Path to the CCSDS data packet file.
+
+    Returns
+    -------
+    list
+        List of all the unpacked data
+    """
+    # TODO: update path to use a combined packets xtce file
+    xtce_file = imap_module_directory / "hit/packet_definitions/P_HIT_HSKP.xml"
+    logger.info(f"Unpacking {packet_file} using xtce definitions in {xtce_file}")
+    unpacked_packets = decom.decom_packets(packet_file, xtce_file)
+    logger.info(f"{packet_file} unpacked")
+    return unpacked_packets
+
+
+def group_data(unpacked_data: list):
+    """Group data by apid.
+
+    Parameters
+    ----------
+    unpacked_data : list
+        packet list
+
+    Returns
+    -------
+    dict
+        grouped data by apid
+    """
+    logger.info("Grouping packet values for each apid")
+    grouped_data = utils.group_by_apid(unpacked_data)
+
+    # Create data classes for each packet
+    for apid in grouped_data.keys():
+        if apid == HitAPID.HIT_HSKP:
+            logger.info(f"Grouping housekeeping packets - APID: {apid}")
+            for i, packet in enumerate(grouped_data[apid]):
+                # convert data to data classes
+                grouped_data[apid][i] = Housekeeping(packet, "0.0", "hskp_sample.ccsds")
+        else:
+            raise RuntimeError(f"Encountered unexpected APID [{apid}]")
+
+    logger.info("Finished grouping packet data")
+    return grouped_data
+
+
+def create_datasets(data: dict, skip_keys=None):
```

Comment:
Can you use the global utility function instead of writing your own?
https://github.com/IMAP-Science-Operations-Center/imap_processing/blob/1efb973eb1830c4e3407c84e1199d8b83c1a7899/imap_processing/utils.py#L131

Reply:
I borrowed this code from the global utility function. The global function takes in a list of packets produced by space packet parser. The HIT data is stored as data classes to further decommutate some of the data, so I wasn't able to use the global function as is. If other instruments are also using data classes then maybe this global function could be modified to handle both packet formats

Reply:
GLOWS and MAG do something similar and don't use that global utility function.